sklearn/mixture/bayesian_mixture.py

```diff
@@ -0,0 +1,923 @@
+
+        log_gauss = self.nu_ * (np.sum((self.m_ ** 2 / self.precisions_), 1) -
+                                2 * np.dot(X, (self.m_ / self.precisions_).T) +
+                                np.dot(X ** 2, (1. / self.precisions_).T))
+        log_prob = -.5 * (n_features * np.log(2. * np.pi) - log_lambda +
+                          n_features / self.beta_ + log_gauss)
+
+        # save log_lambda for computing lower bound
+        self._log_lambda = log_lambda
+        return log_prob
+
+    def _estimate_log_prob_spherical(self, X):
+        n_features = X.shape[1]
+
+        log_lambda = n_features * (digamma(self.nu_) -
+                                   np.log(self.precisions_))
+
+        log_gauss = self.nu_ / self.precisions_ * (
+            np.sum((self.m_ ** 2), 1) - 2 * np.dot(X, self.m_.T) +
+            np.sum(X ** 2, 1)[:, np.newaxis])
+        log_prob = -.5 * (n_features * np.log(2. * np.pi) - log_lambda +
+                          n_features / self.beta_ + log_gauss)
+
+        # save log_lambda for computing lower bound
+        self._log_lambda = log_lambda
+        return log_prob
+
+    def _estimate_lower_bound(self, log_prob, resp, log_resp):
+        """Estimate the lower bound of the model to check the convergence."""
+        # Equation 7.5, 7.6
+        log_p_XZ = np.sum(log_prob * resp)
+        # Equation 7.7
+        log_p_weight = (self._log_dirichlet_norm_prior +
+                        (self._alpha_prior - 1.) * np.sum(self._log_pi))
+        log_p_lambda = self._estimate_p_lambda()
+
+        # Equation 7.10
+        log_q_z = np.sum(resp * log_resp)
+        # Equation 7.11
+        log_q_weight = (np.sum((self.alpha_ - 1.) * self._log_pi) +
+                        log_dirichlet_norm(self.alpha_))
+        log_q_lambda = self._estimate_q_lambda()
+
+        return (log_p_XZ + log_p_weight + log_p_lambda -
+                log_q_z - log_q_weight - log_q_lambda)
+
+    def _estimate_p_lambda(self):
+        return {'full': self._estimate_p_lambda_full,
+                'tied': self._estimate_p_lambda_tied,
+                'diag': self._estimate_p_lambda_diag,
+                'spherical': self._estimate_p_lambda_spherical
+                }[self.covariance_type]()
+
+    def _estimate_p_lambda_full(self):
+        n_features = self._m_prior.shape[0]
+        # Equation 7.9
+        temp1 = (self._log_lambda -
+                 n_features * self._beta_prior / self.beta_)
+
+        mk_sol = np.empty(self.n_components)
+        for k in range(self.n_components):
+            sol = linalg.solve_triangular(
+                self._precisions_chol[k],
+                (self.m_[k] - self._m_prior).T, lower=True).T
+            mk_sol[k] = np.sum(np.square(sol))
+        temp1 -= self._beta_prior * self.nu_ * mk_sol
+
+        # XXX @xuewei4d I think you forgot n_component in your code ?
+        temp1 = (.5 * np.sum(temp1) +
+                 self.n_components * self._log_gaussian_norm_prior)
+
```

Comment:
@xuewei4d I think you forgot to multiply the `log_gaussian_norm` by `n_components`. Could you confirm it for the 4 functions please ?